src/PhpDoc/TypeNodeResolver.php

```diff
@@ -525,6 +525,20 @@ private function resolveGenericTypeNode(GenericTypeNode $typeNode, NameScope $na
 			return new ErrorType();
 		} elseif ($mainTypeName === 'value-of') {
 			if (count($genericTypes) === 1) { // value-of<ValueType>
+				if ($genericTypes[0] instanceof ObjectType) {
+					$classReflection = $genericTypes[0]->getClassReflection();
```

Comment:
Please check `$this->getReflectionProvider()->hasClass` + `$this->getReflectionProvider()->getClass` like here instead: https://github.com/phpstan/phpstan-src/blob/8ba5f3499f166c2a4e2d80f2f09d929e6f598ee5/src/PhpDoc/TypeNodeResolver.php#L542-L546